java/src/jmri/BlockManager.java

```diff
@@ -144,19 +144,21 @@ public Block createNewBlock(@Nonnull String userName) {
      *
      * @param name the system name or the user name for the block
      * @return a new or existing Block
-     * @throws IllegalArgumentException if cannot create block; never returns
-     *                                  null
+     * @throws IllegalArgumentException if cannot create block or no name supplied; never returns null
      */
     @Nonnull
     public Block provideBlock(@Nonnull String name) {
+        if (name.isEmpty()) {
+            throw new IllegalArgumentException("Could not create block, no name supplied");
```

Comment:
Thanks for the PR.
Please change `IllegalArgumentException` to `BadSystemNameException` or `BadUserNameException`.

Reply:
`BadNameException` is a better choice than `IllegalArgumentException`.
`BadNameException` is the parent class of `BadSystemNameException` and `BadUserNameException`.